### PR TITLE
fix(forms-web-app): validate the payload to create or update appeal

### PR DESCRIPTION
**Pull request by a6software**
## Ticket Number

UCD-944

## Description of change

Bit of a proof of concept - all the controller need this, so I am doing them all. This is the first.

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: forms-web-app/tests/unit/controllers/appellant-submission/site-access-safety.test.js [hunk 1]
```diff
@@ -92,35 +105,79 @@ describe('controllers/appellant-submission/site-access-safety', () => {
 
       expect(res.redirect).toHaveBeenCalledWith(`/${VIEW.APPELLANT_SUBMISSION.TASK_LIST}`);
 
-      const { empty: appeal } = APPEAL_DOCUMENT;
-      appeal[sectionName][taskName].hasIssues = true;
```
Comment (a6software) [inline forms-web-app/tests/unit/controllers/appellant-submission/site-access-safety.test.js:96]: it turns out you can change any of these values and the tests still pass :exploding_head:

File: forms-web-app/tests/unit/controllers/appellant-submission/site-access-safety.test.js [hunk 2]
```diff
@@ -5,18 +5,29 @@ const logger = require('../../../../src/lib/logger');
...
 describe('controllers/appellant-submission/site-access-safety', () => {
+  let req;
+  let res;
+  let appeal;
+
+  beforeEach(() => {
+    req = mockReq();
+    res = mockRes();
+
+    ({ empty: appeal } = APPEAL_DOCUMENT);
```
Comment (a6software) [inline forms-web-app/tests/unit/controllers/appellant-submission/site-access-safety.test.js:26]: TIL: reassigning via destructuring ... :boom: 

tenor

Comment (danWatkinson) [reply forms-web-app/tests/unit/controllers/appellant-submission/site-access-safety.test.js:26]: lol - starting the new year with enthusiasm?

**Review by danWatkinson — APPROVED** [review]
lgtm

**Comment by a6software** [comment]
ty

**Pull request merged**